tools/jdbc/src/test/java/org/duckdb/test/TestDuckDBJDBC.java

```diff
@@ -2318,6 +2318,154 @@ public static void test_appender_date_and_time() throws Exception {
         stmt.close();
         conn.close();
     }
+
+    public static void test_appender_decimal() throws Exception {
+        DuckDBConnection conn = DriverManager.getConnection("jdbc:duckdb:").unwrap(DuckDBConnection.class);
+        Statement stmt = conn.createStatement();
+
+        stmt.execute(
+            "CREATE TABLE decimals (id INT4, a DECIMAL(4,2), b DECIMAL(8,4), c DECIMAL(18,6), d DECIMAL(38,20))");
+        DuckDBAppender appender = conn.createAppender(DuckDBConnection.DEFAULT_SCHEMA, "decimals");
+
+        BigDecimal bd16 = new BigDecimal("12.34").setScale(2);
```

Comment:
Can you break this test up a bit? And use slightly more comprehensible variable names?